src/compiler/propagation/type_primitive_esp32.cc

```diff
@@ -40,6 +40,10 @@ TYPE_PRIMITIVE_ANY(memory_page_report)
 TYPE_PRIMITIVE_ANY(watchdog_init)
 TYPE_PRIMITIVE_ANY(watchdog_reset)
 TYPE_PRIMITIVE_ANY(watchdog_deinit)
+TYPE_PRIMITIVE_ANY(pin_hold_enable)
```

Comment:
These really only return null.

Reply:
Agreed.
There wasn't any 'NULL' variant in the file and I didn't look for one. I figured that nobody would use the result anyway.
Changed the watchdog and pin-hold functions to be typed as null.
done.